### PR TITLE
A possible fix about esy

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

File: install/install.ml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env ocaml
 
-#use "topfind" ;;
-#require "unix" ;;
+#load "unix.cma" ;;
 
 let xen = "xen_linkopts = \"-l:laolao/xen/liblaolao_xen_stubs.a\""
 let freestanding = "freestanding_linkopts = \"-l:laolao/freestanding/liblaolao_freestanding_stubs.a\""
-let meta = "_build/default/META.checkseum"
+let meta = match Sys.getenv_opt "DUNE_BUILD_DIR" with
```
Comment (jordwalke) [inline install/install.ml:7]: This looks good.

File: checkseum.opam [hunk 2]
```diff
@@ -40,7 +40,7 @@ depends: [
   "fmt"           {with-test}
   "fpath"         {with-test}
   "rresult"       {with-test}
-  "ocamlfind"     {build & with-test}
+  "ocamlfind"     {with-test}
```
Comment (jordwalke) [inline checkseum.opam:43]: I mentioned on [the thread](https://github.com/mirage/checkseum/issues/48) and my understanding is that the `with-test` is actually what is causing `ocamlfind` to not be considered a dependency, when it actually is required to build/install checkseum. So I would have thought that the fix is to remove `with-test`.

It is required to build/install checkseum even when not just building it to run tests, because the install step uses `topfind`
https://github.com/mirage/checkseum/blob/14b524e255cc38583c7749924b848b4215343641/install/install.ml#L3

`with-test` is documented in the [opam docs](https://opam.ocaml.org/doc/Manual.html) with:
> with-test: only true if tests have been enabled for this specific package
> with-test dependencies are only needed when building tests (when the package is explicitly installed with --with-test)

When consumed regularly, the package will not be installed with the `--with-test` flag so the predicate evaluates to false and ocamlfind is not effectively a dependency. For that same reason, I _suspect_ the line in `install:` section also needs to change to remove `with-test` but I'm not sure.

```
  [ "dune" "install" "-p" name ] {with-test}
```

Comment (dinosaure) [reply checkseum.opam:43]: Currently, this PR deletes the `ocamlfind` dependency to build/install `checkseum`:
https://github.com/mirage/checkseum/blob/1cdf0117992b25ca19c3e81b635ff26d5f8aa492/install/install.ml#L3

Then, about the `dune` installation tagged with `with-test`, it's a corner case to see if `checkseum` respects the layout needed by MirageOS. The behavior of OPAM is to follow the `checkseum.install` generated by `dune` (it does not call `dune build @install`). If `esy` does the same (independently what the OPAM file describes about `install`), this PR should be fine for both.

Comment (jordwalke) [reply checkseum.opam:43]: Whoops, I missed that part where you removed the actual ocamlfind dependency. I think this should all work then. Yeah, esy will respect the `checksum.install` if present automatically like opam (In general we try to make esy be totally opam compliant). The only possible difference is that it sandboxes each package's build so that it can't rely on dependencies that weren't actually specified and we end up catching a lot of package configuration issues that broke out of their dependency sandbox undetected. I'd like to find a way to catch these things before PRs are merged to opam-repository - maybe some kind of a service that checks each opam-repository PR and tries to depend on it and cross references the PR if there's an issue building with esy but not opam.

 I appreciate the fix.

**Comment by dinosaure** [comment]
Error is unrelated, ready to merge.
